pyinstrument/__main__.py

```diff
@@ -44,64 +45,74 @@ def main():
         dest='color', action='store_false',
         help='force no color text output')
 
+    parser.add_option('-m', '',
+        dest='module_name', action='store',
+        help='searches sys.path for the named module and runs the  corresponding .py file as a script.')
+
     if not sys.argv[1:]:
         parser.print_help()
         sys.exit(2)
 
-    (options, args) = parser.parse_args()
-    sys.argv[:] = args
+    options, args = parser.parse_args()
 
-    if len(args) > 0:
-        progname = args[0]
-        sys.path.insert(0, os.path.dirname(progname))
+    if args == [] and options.module_name is None:
+        parser.print_help()
+        sys.exit(2)
 
-        with open(progname, 'rb') as fp:
-            code = compile(fp.read(), progname, 'exec')
-        globs = {
-            '__file__': progname,
-            '__name__': '__main__',
-            '__package__': None,
-        }
+    sys.argv[:] = args
 
-        if options.output_renderer:
-            renderer = options.output_renderer
-        elif options.output_html:
-            renderer = 'html'
-        else:
-            renderer = 'text'
+    if options.module_name is not None:
+        progname = find_spec(options.module_name).origin
```

Comment:
I'm not sure this is right - when I run `find_spec('pyinstrument').origin` in a shell it gives me the `__init__.py` file. It should be running the `__main__.py` file as I understand it.

The [cProfile module](https://github.com/python/cpython/blob/master/Lib/cProfile.py#L157) uses [runpy](https://docs.python.org/2.7/library/runpy.html) for this - it's probably best to copy that.